lib/ansible/modules/packaging/os/rhn_register.py

```diff
@@ -297,7 +297,9 @@ def unregister(self):
         # Remove systemid file
         os.unlink(self.config['systemIdPath'])
 
-    def subscribe(self, channels):
+    def subscribe(self, channels=[]):
```

Comment:
Not sure I understand this. 

This makes channels a mutable default arg, meaning the same reference will be used for all calls of this method which is almost always the wrong behavior.

Why subscribe() need a default arg? it is only ever called from once place in module main, and it is called with the 'channels' var which is a list by default. How would subscribe() get called without channels being a list?

Reply:
@alikins I agree with you about the redundant initialization of channels in the subscribe method declaration. However at least in my case, the `if len(channels) <= 0:` statement at the beginning of this function is solving the bug mentioned here, in which the play would fail with the `cannot marshal None unless..` exception. Is this going go be included in the next release? 
Thanks!

Reply:
If it's valid to call `subscribe` with an empty list or `None` for channels, use this instead:

```python
def subscribe(self, channels):
    if not channels:
         return
```

Reply:
@mattclay yep I agree, seems like the right way to do it!
So who can change the PR? I see it was opened by some kind of a bot? (WhyIsThisOpen).
I don't mind raising the glove and creating a new PR with the change you suggested, so that we'll have it included in the next release.

Reply:
I am not a bot.